client/src/ui/molecules/offline-status-bar/index.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from "react";
+
+import { isPayingSubscriber } from "../../../utils";
+import { useOnlineStatus } from "../../../hooks";
+import { useUserData } from "../../../user-context";
+
+import "./index.scss";
+
+export function OfflineStatusBar() {
+  const user = useUserData();
+  const onlineStatus = useOnlineStatus();
+  const statusBarRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    const statusBar = statusBarRef.current;
+    const handleOnline = () => {
+      statusBar?.classList.add("is-online");
+    };
+
+    window.addEventListener("online", handleOnline);
+
+    if (statusBar) {
+      // when the fade-out animation ends, remove the class so
+      // the statusbar is hidden not just faded out.
+      statusBar.addEventListener("animationend", () => {
+        statusBar.classList.remove("is-online");
+      });
+    }
+
+    return () => {
+      window.removeEventListener("online", handleOnline);
+    };
+  }, [onlineStatus]);
```

Comment:
The hook allows us to simplify this a lot:

```suggestion
  const { isOnline } = useOnlineStatus();
  const statusBarRef = React.useRef<HTMLDivElement>(null);

  React.useEffect(() => {
    const statusBar = statusBarRef.current;
    statusBar?.classList.toggle("is-online", isOnline);
  }, [isOnline]);
```

Reply:
So, I have this:

```js
React.useEffect(() => {
    const statusBar = statusBarRef.current;
    const handleOnline = () => {
      statusBar?.classList.add("is-online");
    };

    window.addEventListener("online", handleOnline);

    statusBar?.addEventListener("animationend", () => {
      statusBar.classList.remove("is-online");
    });

    return () => {
      window.removeEventListener("online", handleOnline);
    };
  }, []);
```

For the following reasons:

1. I need to know when the user is back online so that I can add the class `is-online`. This does a fade-out animation of the bar but, it does not completely remove the bar at the end of the animation so,
2. I need the `animationend` listener so that I remove the class once the animation is done. This then returns the bar to the `display: none;` state.
3. I create the `handleOnline` function so I can remove it in the `useEffect` cleanup

Reply:
@schalkneethling Would't this still work:

```
React.useEffect(() => {
  const statusBar = statusBarRef.current;
  if (isOnline && statusBar) {
    statusBar.classList.add("is-online");
    statusBar.addEventListener("animationend", () => {
      statusBar.classList.remove("is-online");
    });
  };
}, [isOnline]);
```

Reply:
Also, we should probably set the `{ once: true }` option on the `addEventListener` call.